frame.js

```diff
@@ -0,0 +1,50 @@
+class Frame {
+  constructor(rolls) { //constructed with rolls e.g. new Frame([3,3])
+    this.rolls = rolls 
+  }
+
+  _isStrike() {
+    return this.rolls[0] == 10; 
+  }
+  _isSpare() {
+    return this._sumFrame() == 10;
+  }
+  _isFinalFrame(next_frame, next_next_frame) {
+    return next_frame == undefined && next_next_frame == undefined;
+  }
+  score(next_frame, next_next_frame) {
+    //console.log(this._isFinalFrame(next_frame, next_next_frame))
+    if (this._isFinalFrame(next_frame, next_next_frame)) { 
+      //console.log(this._sumFrame());
+        return this._sumFrame();
+    }
+    else {
+      return this._sumFrame() + this._bonus(next_frame, next_next_frame);
+    }
+  }
+
+  _bonus(next_frame, next_next_frame) { //frame class does not need to know what next_frame and next_next_frame are, because these params will be passed in when scorecard methods are called
+    if (this._isStrike() && next_frame._isStrike()) {
+      if (next_next_frame == undefined) { // i.e. 9th frame
+        return 10 + next_frame.rolls[1];
+        } 
+      else {
+        return 10 + next_next_frame.rolls[0];
+      }
+    }
```

Comment:
That works. Something else you could do here it to flatten the next and next next and then take two bonuses - you wouldn't need to know where they come from.